lib/proxy/provider.rb

```diff
@@ -0,0 +1,30 @@
+class ::Proxy::Provider
+  include ::Proxy::Pluggable
+  include ::Proxy::Log
+
+  class << self
+    attr_reader :provider_factory_proc
+
+    def plugin(plugin_name, aversion, attrs)
```

Comment:
Should this class inherit from Proxy::Plugin in some way? https://github.com/theforeman/smart-proxy/pull/256/files#diff-cc107af997731be1551c20e4af1d19b2R114 shows that aside from provider_factory this class is effectively the same.